msteams-platform/messaging-extensions/high-quality-message-extension.md

```diff
@@ -0,0 +1,511 @@
+---
+title: Bot-based message extension plugin guidelines
+description: Learn about the guidelines and criteria to extend your message extension as a plugin for Copilot.
+ms.topic: how-to
+author: v-preethah
+ms.author: surbhigupta
+ms.localizationpriority: high
+ms.date: 10/18/2023
+---
+
+# Guidelines to create or upgrade a message extension plugin for Copilot
+
+Microsoft 365 plugins provide integration with various Microsoft hubs, such as Teams and Outlook. The integration helps users to search or create content in external systems. Message extension plugins allow Copilot to interact with APIs from other software and services through a bot. You can:
+
+* Search for the latest information or record, for example, the latest incident ticket or survey results.
+* Summarize information based on multiple records. For example, summarize all incidents tickets related to the project Northwind.
+
+We recommend that you build or upgrade your existing message extensions to get the best out of Copilot. To ensure Copilot recognizes your app, build message extensions that can at least, search or summarize information.   Additionally, your extensions should meet the standards for compliance, performance, security, and user experience outlined in this article.
```

Comment:
> To ensure Copilot recognizes your app, build message extensions that can at least, search or summarize information.

How about: Message extensions should support one or more search commands, as these are recognized by Copilot as skills it can execute on behalf of the user.

Reply:
Updated as Message extensions should support one or more search commands, as these are recognized by Copilot as skills it can execute on behalf of the user.